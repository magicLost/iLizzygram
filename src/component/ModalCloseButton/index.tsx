import React from "react";
import IconButton from "../IconButton";
import CloseIcon from "../Icons/CloseIcon";
import classes from "./ModalCloseButton.module.scss";

interface ModalCloseButtonProps {
  onClick: (event: any) => void | undefined;
  color: TColorProp;
  ariaLabel: string;
}

/* const useStyles = makeStyles({
  root: {
    position: "absolute",
    top: "0",
    right: "0",
    zIndex: 1315,
  },
}); */

const ModalCloseButton = ({
  onClick,
  color,
  ariaLabel,
}: ModalCloseButtonProps) => {
  //const classes = useStyles();
  return (
    <div className={classes.root}>
      <IconButton
        icon={<CloseIcon width={16} height={16} color={color} />}
        onClick={onClick}
        ariaLabel={ariaLabel}
      />
    </div>
  );
};

export default ModalCloseButton;
