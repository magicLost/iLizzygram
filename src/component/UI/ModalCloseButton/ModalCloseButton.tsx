import React from "react";
//import classes from './ModalCloseButton.module.scss';
import { makeStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";

interface ModalCloseButtonProps {
  onClick: (event: any) => void | undefined;
  closeIconSize?: "large" | "small" | "default" | "inherit";
  ariaLabel: string;
}

const useStyles = makeStyles({
  root: {
    position: "absolute",
    top: "0",
    right: "0",
    zIndex: 1315,
  },
});

const ModalCloseButton = ({
  onClick,
  ariaLabel,
  closeIconSize = "default",
}: ModalCloseButtonProps) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <IconButton onClick={onClick} color="secondary" aria-label={ariaLabel}>
        <CloseIcon fontSize={closeIconSize} />
      </IconButton>
    </div>
  );
};

export default ModalCloseButton;
