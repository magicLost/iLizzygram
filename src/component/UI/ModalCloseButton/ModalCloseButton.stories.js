import React from "react";
import { action } from "@storybook/addon-actions";
import ModalCloseButton from ".";

export default {
  component: ModalCloseButton,
  title: "Buttons/ModalCloseButton",
  decorators: [],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

export const Default = () => {
  return (
    <>
      <ModalCloseButton
        onClick={() => console.log("onClick")}
        ariaLabel="close something"
        closeIconSize="default"
      />
    </>
  );
};
