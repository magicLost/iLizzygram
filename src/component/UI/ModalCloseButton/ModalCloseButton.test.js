import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import ModalCloseButton from ".";
import classes from "./ModalCloseButton.module.scss";

describe("ModalCloseButton", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {
      _render = render(<ModalCloseButton />);
    });

    afterEach(cleanup);

    describe("Snapshots", () => {
      test("matches snapshot", () => {
        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
