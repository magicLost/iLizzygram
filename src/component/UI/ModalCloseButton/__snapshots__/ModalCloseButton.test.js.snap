// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCloseButton Render and props test Snapshots matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
